packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -108,9 +109,13 @@ List<Map<String, Object>> _filterPluginsByPlatform(List<Plugin> plugins, String
   final Set<String> pluginNames = platformPlugins.map((Plugin plugin) => plugin.name).toSet();
   final List<Map<String, Object>> pluginInfo = <Map<String, Object>>[];
   for (final Plugin plugin in platformPlugins) {
+    // This is guaranteed to be non-null due to the `where` filter above.
+    final PluginPlatform platformPlugin = plugin.platforms[platformKey]!;
     pluginInfo.add(<String, Object>{
       _kFlutterPluginsNameKey: plugin.name,
       _kFlutterPluginsPathKey: globals.fsUtils.escapePath(plugin.path),
+      if (platformPlugin is NativeOrDartPlugin)
+        _kFlutterPluginsHasNativeBuildKey: (platformPlugin as NativeOrDartPlugin).isNative(),
```

Comment:
I see. `isNative()` is confusing. I think, it should have been called `hasPlatformImplementation()`.  I'm not sure if you like the short version instead.

Reply:
`hasPlatformImplementation` seems like a super confusing name to me, given that "platform" in all of this code refers to the plugin platform key. E.g., under that naming `platformPlugin.hasPlatformImplementation()` would return false for `path_provider_windows` for the `windows` platform (because it's Dart-only). But it obviously does have a platform implementation for Windows, because that's what the package is.

Reply:
would it work if we instead have `isDartOnly()`?

Reply:
There are some changes coming to some of this naming in the PR to add FFI plugin templates; we should probably address this there instead since the names become more nuanced there.

The current proposal there replaces this with:
```
  /// Determines whether the plugin has a Dart implementation.
  bool hasDart();

  /// Determines whether the plugin has a FFI implementation.
  bool hasFfi();

  /// Determines whether the plugin has a method channel implementation.
  bool hasMethodChannel();
```

which I think will address your concern.